src/Client.php

```diff
@@ -32,35 +34,46 @@ class Client
     /**
      * @var array
      */
-    protected $httpHeaders;
+    protected $options;
 
     /**
      * @var string
      */
     protected $requestMethod;
 
+    /**
+     * @var AuthInterface
+     */
+    protected $auth;
+
     /**
      * Client constructor.
      *
      * @param string $endpointUrl
      * @param array $authorizationHeaders
      * @param array $httpOptions
-     * @param ClientInterface $httpClient
+     * @param ClientInterface|null $httpClient
      * @param string $requestMethod
+     * @param AuthInterface|null $auth
+     * @param array $options
      */
     public function __construct(
         string $endpointUrl,
         array $authorizationHeaders = [],
         array $httpOptions = [],
         ClientInterface $httpClient = null,
-        string $requestMethod = 'POST'
+        string $requestMethod = 'POST',
+        AuthInterface $auth = null,
+        array $options = []
     ) {
         $headers = array_merge(
             $authorizationHeaders,
             $httpOptions['headers'] ?? [],
             ['Content-Type' => 'application/json']
         );
-
+        $this->auth = $auth ?? new HeaderAuth();
+        $this->options = $options;
+        $this->options['headers'] = $headers;
```

Comment:
I don't like that we have a new argument for the constructor `options` and that we renamed `httpHeader` class variable to `options` this makes things very complicated and confusing.
Why don't we just have the user pass an extra value to the `httpOptions` array they already use, and make the array key more specific to be `aws_region" instead of just "region"?

Reply:
I've changed code in the way you described.
One thing - `aws_region` parameter has nothing to do with $httpOptions, I would say it's just an $option. So, maybe we can rename $httpOptions -> $options to be more consistend? (I would be completely fine if we leave it as it is)

Reply:
I don't disagree with you on this. The reason why I want to leave this as it is is because I already have a v2.0 under dev that will overhaul how this works and I want to include your feature in this release. So if we don't modify the current setup a lot, it makes rebasing my work easier. In other words, I'd rather not introduce too many changes unless they're critical. Hope that makes sense.

Reply:
I understand, it makes sense.